### DISCUSSION TITLE
Dynamic expression columns

### DISCUSSION THREAD

**Discussion by haithium** [saulpw/visidata#2289] [Q&A] [answered]
When I create an expression column and reopen VisiData, the expression column turns into a static value, and no updates occur. Is there a way to keep the expression dynamic?"

**Comment by justin2004** [comment]
>When I create an expression column and reopen VisiData

@mdSlash do you mean when you create an expression column, save the file as csv, then open the csv file in visidata...?

**Reply by haithium** [reply]
yes

**Reply by justin2004** [reply]
to do that i think visidata would need it's own file format. i don't think it has one right now though.

**Reply by anjakefala** [reply]
Try playing with [.vds](https://www.visidata.org/docs/internal_formats/)! Try using `gCtrl+S`, and save your sheets using the `.vds` extension. Let us know how it goes.

**Reply by justin2004** [reply]
oh cool, @anjakefala ! i didn't know about that.

**Reply by haithium** [reply]
I don't use VisiData anymore. My first goal is to play around with expressions and create dynamic things. I tried to skim the docs or find help before spending time to deep dive into it. @anjakefala After your comment, I tried saving the file as ".vd" and applied a simple expression `colA > 100`. It worked fine until I saved the file, closed VisiData, and reopened the same file. the expression column changed from dynamic to static.

**Reply by anjakefala** [reply, marked as answer]
I just tested it, and it worked for me! Did you save as `.vds`? The file format in question is `.vds`, not `.vd`.

**Reply by anjakefala** [reply]
We're actively working on improving our documentation: https://github.com/saulpw/visidata/issues/2313 Everyone is welcome to help out.

**Reply by haithium** [reply]
Now it's working after I saved it as `.vds`. Thanks for your help
